pkg/networkservice/common/clusterinfo/server.go

```diff
@@ -19,27 +19,43 @@ package clusterinfo
 
 import (
 	"context"
-	"io/ioutil"
+	"sync/atomic"
 
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 	"gopkg.in/yaml.v2"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/fs"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
 )
 
 type clusterinfoServer struct {
-	configPath string
+	configPath        string
+	clusterInfoSource atomic.Value
```

Comment:
Would it perhaps make sense to consider go 1.19 atomic.Pointer[T]:

https://pkg.go.dev/sync/atomic#Pointer